packages/imperative/src/censor/src/Censor.ts

```diff
@@ -0,0 +1,340 @@
+     */
+    public static setCensoredOptions(censorOpts?: ICensorOptions) {
+        this.censored_options = new Set(this.DEFAULT_CENSORED_OPTIONS);
+
+        if (censorOpts) {
+            // Save off the config object
+            this.mConfig = censorOpts.config;
+
+            // If we have a ProfileTypeConfiguration (i.e. ImperativeConfig.instance.loadedConfig.profiles)
+            if (censorOpts.profiles) { this.setProfileSchemas(censorOpts.profiles); }
+
+            for (const profileType of this.profileSchemas ?? []) {
+                // If we know the command we are running, and we know the profile types that the command uses
+                // we should only use those profiles to determine what should be censored. If we do not, we should
+                // add everything
+                if (censorOpts.commandDefinition == null ||
+                    censorOpts.commandDefinition.profile?.optional?.includes(profileType.type) ||
+                    censorOpts.commandDefinition.profile?.required?.includes(profileType.type)) {
+                    this.handleSchema(profileType);
+                }
+            }
+
+            // Include any secure options from the config
+            if (censorOpts.config) {
+                const secureOptions: Set<string> = new Set();
+
+                // Try to use the command and inputs to find the profiles being loaded
+                if (censorOpts.commandDefinition && censorOpts.commandArguments) {
+                    const profiles = [];
+                    for (const prof of censorOpts.commandDefinition.profile?.required || []) {
+                        profiles.push(prof);
+                    }
+                    for (const prof of censorOpts.commandDefinition.profile?.optional || []) {
+                        profiles.push(prof);
+                    }
+
+                    for (const prof of profiles) {
+                        // If the profile exists, append all of the secure props to the censored list
+                        let profName = censorOpts.commandArguments?.[`${prof}-profile`];
+                        if (!profName) { profName = this.mConfig.mProperties.defaults[`${prof}`]; }
+                        if (profName && censorOpts.config.api.profiles.get(profName)) {
+                            censorOpts.config.api.secure.securePropsForProfile(profName).forEach(prop => secureOptions.add(prop));
+                        }
+                    }
+                } else {
+                    // We only have a configuration file, assume every property that is secured should be censored
+                    censorOpts.config.api.secure.findSecure(censorOpts.config.mProperties.profiles, "profiles").forEach(
+                        prop => secureOptions.add(prop.split(".").pop())
+                    );
+                }
+                secureOptions.forEach(prop => this.addCensoredOption(prop));
+            }
```

Comment:
SonarCloud flagged the `setCensoredOptions` function for complexity. I was wondering if we could move this part of the function into a helper function to reduce complexity? I think it would make it a bit easier to follow the different branches of logic, while also resolving the SonarCloud issue as a result.

Reply:
We could do that, though the data would need to be passed through to the second helper function, and it would probably only be called in this function. I attempted to do some reduction of calling `addCensoredOption` in here, which might not be as easily possible if I break this up into multiple functions.